### PR TITLE
ref(ts) Convert CheckboxFancy to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx [hunk 1]
```diff
@@ -18,7 +25,7 @@ const getHoverStyles = p =>
     border: 2px solid ${p.checked ? p.theme.purple : p.theme.gray4};
   `;
 
-const CheckboxFancy = styled(({checked, disabled, ...props}) => (
+const CheckboxFancy = styled(({checked, disabled, ...props}: Props) => (
   <div role="checkbox" aria-disabled={disabled} aria-checked={checked} {...props}>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/checkboxFancy.tsx]: Looks like we spread into a div, this should probably include `HtmlDivElement` props right?

**Review by evanpurkhiser — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/checkboxFancy.tsx]: We never actually use the spread for anything productive. I'm going to remove the spread and pass the className prop down instead.

**Comment by lynnagara** [comment]
CheckboxFancy. While we're here, do you think it's worth renaming, making it a variant of checkbox or even just moving it directly into the one component where it's used?

**Comment by markstory** [comment]
@lynnagara I can join it into its parent component as a separate pull request. This checkbox also has storybook file that would need to be removed.

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx [hunk 2]
```diff
@@ -5,6 +5,13 @@ import {css} from '@emotion/core';
 import PropTypes from 'prop-types';
 import InlineSvg from 'app/components/inlineSvg';
 
+type Props = {
+  disabled?: boolean;
+  checked?: boolean;
+  size?: string;
+  className?: string;
+} & HTMLDivElement;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/checkboxFancy.tsx]: Is this right? I think it's like, `React.HTMLAttributes<HTMLDivElement>` or something like that

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx [hunk 3]
```diff
@@ -18,8 +25,13 @@ const getHoverStyles = p =>
...
 
-const CheckboxFancy = styled(({checked, disabled, ...props}) => (
-  <div role="checkbox" aria-disabled={disabled} aria-checked={checked} {...props}>
+const CheckboxFancy = styled(({checked, disabled, className}: Props) => (
+  <div
+    role="checkbox"
+    aria-disabled={disabled}
+    aria-checked={checked}
+    className={className}
+  >
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/checkboxFancy.tsx]: Oh we don't spread anymore now either, so I guess we don't need that haha

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx [hunk 4]
```diff
@@ -18,8 +25,13 @@ const getHoverStyles = p =>
     border: 2px solid ${p.checked ? p.theme.purple : p.theme.gray4};
   `;
 
-const CheckboxFancy = styled(({checked, disabled, ...props}) => (
-  <div role="checkbox" aria-disabled={disabled} aria-checked={checked} {...props}>
+const CheckboxFancy = styled(({checked, disabled, className}: Props) => (
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/checkboxFancy.tsx]:
```suggestion
const CheckboxFancy = styled('div')<Props>(({checked, disabled, className}) => (
```

Comment (markstory) [reply src/sentry/static/sentry/app/components/checkboxFancy.tsx]: Will that add an additional div element into the DOM?

File: src/sentry/static/sentry/app/components/checkboxFancy.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/checkboxFancy.tsx]: Okay

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**